### DISCUSSION TITLE
how to  change font family of the theme

### DISCUSSION THREAD

**Discussion by markaguilar** [heroui-inc/heroui#908] [Help] [answered]
how to  change font family of the theme

**Comment by tianenpang** [comment, marked as answer]
Hi @markaguilar you can customize the fonts by `globalStyles` and `createTheme`.

- Specify the `@font-face` in [`globalStyles`](https://nextui.org/docs/theme/customize-theme#global-styles).
- Overwrite or add `fonts` token in [`createTheme`](https://nextui.org/docs/theme/customize-theme#customizing-theme-tokens).
- Next.js [preload](https://leerob.io/blog/fonts#preloading) self-hosted font in [`pages/_document.tsx`](https://nextjs.org/docs/advanced-features/custom-document) or use [`@next/font`](https://nextjs.org/docs/api-reference/next/font).

Here's an example with Next.js: [codesandbox.io/s/nextui-discussion-908-custom-fonts-6y5yhk](https://codesandbox.io/s/nextui-discussion-908-custom-fonts-6y5yhk)

**Reply by armatrix** [reply]
so, how to override all variants' font family?  or I want to make 'h1', 'h2', 'h3' use one font family,  and using another font for others, what should I do? Is there any method like override 'Text:{h1:{fontFamily:"some font"}}'?

**Reply by tianenpang** [reply]
You can use the tag selector to override the styles in [`globalCss`](https://nextui.org/docs/theme/customize-theme#global-styles).

Example with [`next/font`](https://nextjs.org/docs/api-reference/next/font) -  [codesandbox.io/p/sandbox/nextui-discussion-908-custom-fonts-2-ub3814](https://codesandbox.io/p/sandbox/nextui-discussion-908-custom-fonts-2-ub3814)
